raster/r.neighbors/readweights.c

```diff
@@ -12,33 +13,55 @@ void read_weights(const char *filename)
 
     ncb.weights = G_malloc(ncb.nsize * sizeof(DCELL *));
     for (i = 0; i < ncb.nsize; i++)
-	ncb.weights[i] = G_malloc(ncb.nsize * sizeof(DCELL));
+        ncb.weights[i] = G_malloc(ncb.nsize * sizeof(DCELL));
 
     if (!fp)
-	G_fatal_error(_("Unable to open weights file %s"), filename);
+        G_fatal_error(_("Unable to open weights file %s"), filename);
 
     for (i = 0; i < ncb.nsize; i++)
-	for (j = 0; j < ncb.nsize; j++)
-	    if (fscanf(fp, "%lf", &ncb.weights[i][j]) != 1)
-		G_fatal_error(_("Error reading weights file %s"), filename);
+        for (j = 0; j < ncb.nsize; j++)
+            if (fscanf(fp, "%lf", &ncb.weights[i][j]) != 1)
+                G_fatal_error(_("Error reading weights file %s"), filename);
 
     fclose(fp);
 }
 
-void gaussian_weights(double sigma)
+double gaussian(double factor, double squared_distance)
+{
+    double sigma2 = factor * factor;
+
+    return exp(-squared_distance / (2 * sigma2)) / (2 * M_PI * sigma2);
+}
+
+double exponential(double factor, double squared_distance)
+{
+    return exp(factor * sqrt(squared_distance));
+}
+
+void compute_weights(const char *function_type, double factor)
 {
-    double sigma2 = sigma * sigma;
     int i, j;
+    double (*weight) (double, double);
+
+    if (!strcmp(function_type, "gaussian")) {
+        weight = gaussian;
+    }
+    else if (!strcmp(function_type, "exponential")) {
```

Comment:
This check is redundant as there are only two options of function type at this point ("file" and "none" are already taken care of). Although it does not cause any harm per se, it triggers a warning in code analysis tools. Change to plain `else {`